pkg/sshd/server_test.go

```diff
@@ -186,18 +184,15 @@ func TestConcurrentPushSameRepo(t *testing.T) {
 		case sessOut := <-outCh:
 			output := sessOut.outStr
 			err := sessOut.err
-			if output != multiPushLine && output != "OK" {
-				t.Fatalf("expected 'OK' or '%s', but got '%s' (error '%s')", multiPushLine, output, err)
+			if !foundOK && output == "OK" {
+				continue
 			}
-
-			if sessOut.err != nil {
-				t.Fatalf("found '%s' output with an error '%s'", output, err)
+			if foundOK && err != nil && output != "OK" {
+				assert.Equal(t, output, multiPushLine, "output")
+				continue
```

Comment:
@smothiki you're right - it wasn't catching the inconsistent err messages because `foundOK` was never `true` and this block was never executing - see [above](https://github.com/deis/builder/pull/361#discussion_r66129214). Now that it is, the error is coming up...